### PR TITLE
Adding Background task

### PR DESCRIPTION
**Pull request by ran-helfer**
### Note the base branch

### CODE REVIEW

File: api_endpoints.py [hunk 1]
```diff
@@ -4,10 +4,17 @@
...
+from time import sleep
+import sys
 
 app = Flask(__name__)
 api = Api(app)
 
+SLEEP_TIME = int(1)
+
+if(len(sys.argv)>=2):
+    SLEEP_TIME = int(sys.argv[1])
```
Comment (ran-helfer) [inline api_endpoints.py:16]: We can now run:

`python api_endpoints.py 15
`

So we can check `POST` requests that are taking `15 `seconds at least.

It is good for testing background flow on our client

File: VizNetworkApp/ViewController.swift [hunk 2]
```diff
@@ -71,6 +74,25 @@ class ViewController: UIViewController, UITableViewDataSource {
         }
         
     }
+
+    
+    
+    /* Delta time is about 26-27 seconds */
```
Comment (ran-helfer) [inline VizNetworkApp/ViewController.swift]: This is good info. Apparently all BG tasks are terminated after  **about 26-27 seconds**

**Comment by ran-helfer** [comment]
Screen Shot 2021-08-12 at 16 19 25

I see no leaks with current code version

File: VizNetworkApp/ViewController.swift [hunk 3]
```diff
@@ -14,12 +14,15 @@ class ViewController: UIViewController, UITableViewDataSource {
...
     let reuseIdentifier = "MyUITableViewCellreuseIdentifier"
+    //var backgroundTaskId: UIBackgroundTaskIdentifier? = nil
 
     override func viewDidLoad() {
         super.viewDidLoad()
         tableView.register(UITableViewCell.self, forCellReuseIdentifier: reuseIdentifier)
         tableView.dataSource = self
         getRequest(1)
+        
+        //NotificationCenter.default.addObserver(self, selector: #selector(appMovedToBackground), name: UIApplication.willResignActiveNotification, object: nil)
```
Comment (doviz) [inline VizNetworkApp/ViewController.swift]: Clean

File: VizNetworkApp/ViewController.swift [hunk 4]
```diff
@@ -75,6 +78,25 @@ class ViewController: UIViewController, UITableViewDataSource {
         }
         
     }
+
+    
+    
+    /* Delta time is about 26-27 seconds */
+    
+//    @objc func appMovedToBackground() {
+//        let startTime = Date()
```
Comment (doviz) [inline VizNetworkApp/ViewController.swift]: Clean comments

File: VizNetworkApp/ViewController.swift [hunk 5]
```diff
@@ -14,12 +14,15 @@ class ViewController: UIViewController, UITableViewDataSource {
     /* need to keep reference or decoding won't happen since instance is being released when makeRequest block is finished */
     var usersList: UsersList?
     let reuseIdentifier = "MyUITableViewCellreuseIdentifier"
+    //var backgroundTaskId: UIBackgroundTaskIdentifier? = nil
```
Comment (doviz) [inline VizNetworkApp/ViewController.swift]: clean comments

File: api_endpoints.py (hunk 1 above)
Comment (doviz) [reply api_endpoints.py:16]: cool

**Review by doviz — APPROVED** [review]

**Comment by ran-helfer** [comment]
@doviz - Thanks 🙏🏻 

This is a sample app and our `viewController` code does not go anywhere 🥳

**Pull request merged**